src/views/dashboard/board/Board.js

```diff
@@ -0,0 +1,150 @@
+
+import { AiOutlineDelete, AiOutlineEdit } from 'react-icons/ai';
+import { FaChalkboard } from 'react-icons/fa';
+
+import { Container, Title } from '../team/styles/team';
+
+import ModalBoard from '../../../components/boards/ModalBoard';
+
+import { boardsApi } from '../../../api'
+
+function Board({ form, je }) {
+  const { getFieldDecorator } = form;
+  const [dataSource, setDataSource] = useState([]);
+  const [loading, setLoading] = useState([]);
+
+  useEffect(() => {
+    const loadBoards = async () => {
+      setLoading(true);
+      try {
+        const response = await boardsApi.index(je.id);
+        if (response.status === 200) {
+          const data = response.data.boards.map(board => ({
+            ...board,
+            editable: false
+          }));
+          setDataSource(data);
+        }
+      } catch (error) {
+        console.log(error);
+      }
+      setLoading(false);
+    }
+    loadBoards();
+  }, []);
+
+  function handleAdd(board) {
+    const data = {
+      ...board,
+      editable: false,
+    }
+    setDataSource([...dataSource, data]);
+  }
+
+  async function handleDelete(key) {
+    try {
+      const response = await boardsApi.delete(je.id, { boardId: key });
+      if (response.status === 200) {
+        setDataSource(dataSource.filter(board => board.id !== key));
+        message.success("Diretoria removida com sucesso!");
+      }
+    }
+    catch (error) {
+      message.error(error.response.data.msg);
+    }
+  }
+
+  function handleEdit(board) {
+    board.editable = true;
+    setDataSource([...dataSource]);
+  }
+
+  async function onSaveEdit(board) {
+    try {
+      const response = await boardsApi.update(je.id, {
+        boardId: board.id,
+        name: form.getFieldValue('name'),
+      })
+      if (response.status === 200) {
+        board.name = response.data.board.name;
+        board.editable = false;
+        setDataSource([...dataSource]);
+      }
+    }
+    catch (error) {
+      console.log(error);
+      message.error(error.response.data.msg);
+    }
+  }
+
+  function onCancel(board) {
+    form.resetFields();
+    board.editable = false;
+    setDataSource([...dataSource]);
+  }
+
+  const columns = [
+    {
+      title: 'Nome',
+      dataIndex: 'name',
+      key: 'name',
+      render: (_, record) =>
+        record.editable ? (
+          <Form>
+            <Form.Item>
+              {getFieldDecorator('name', {
+                initialValue: record.name,
+              })(
+                <Input />
+              )}
+            </Form.Item>
+          </Form>
+        ) : <p>{record.name} </p>
+    },
+    {
+      title: 'Deletar ',
+      dataIndex: 'delete',
+      render: (text, record) =>
+        dataSource.length >= 1 ? (
+          <Popconfirm title="Sure to delete?" onConfirm={() => handleDelete(record.id)}>
+            <a><AiOutlineDelete style={{ fontSize: '17pt' }} /></a>
+          </Popconfirm>
+        ) : null,
```

Comment:
Quando vc faz uma verificação usando ternário assim, e depende apenas de um tipo de retorno e caso seja falso não vai interessar vc renderizar nada, vc pode mudar um pouco como faz isso.
No caso, vc depende apenas de uma resposta de true na verificação `dataSource.length >= 1`, e se for falso retorna nada. Em vez de ter que escrever toda a operação ternária e no final acrescentar o null e talz, vc pode fazer:
```javascript
dataSource.length >= 1 && <Popconfirm title="Sure to delete?" onConfirm={() => handleDelete(record.id)}>...</PopConfirm>
```
Assim vc vai obter o mesmo resultado e o código fica menos verboso

Reply:
Entendii, valeus Jp, vou fazer as alterações